### ISSUE TITLE
ForceEqualSignAlignment shouldn't use method definition in alignment

### ISSUE THREAD

**Issue by Dbz** [bbatsov/rubocop#2722]
```
$ rubocop -V
0.36.0 (using Parser 2.3.0.1, running on ruby 2.3.0 x86_64-darwin14)
```

```
# .rubocop.yml
Style/ExtraSpacing:
  ForceEqualSignAlignment: true
```

Running `rubocop -a` turns the following `dfs` method:

```ruby
def dfs(node = @root)
  nodes = []
end
```

into this `dfs` method:

```ruby
def dfs(node = @root)
  nodes      = []
end
```

Rubocop shouldn't take the method definition into consideration when forcing equal sign alignment. I can fix this by adding or modifying the `.select` statement within the `def investigate(processed_source)` method in [extra_spacing.rb](https://github.com/bbatsov/rubocop/blob/8c418ee462e19a63b7bd00bf5b7d4cf84191ceb7/lib/rubocop/cop/style/extra_spacing.rb).

**Comment by jfelchner** [comment]
@alexdowad I think that one of the keys here is that, for reference of proper alignment, the cop needs to go up line by line until it gets to either:

* a line of lesser indentation level as the current line
* a line of the same indentation level as the current line

In that way, both of these (currently broken) cases will be caught:

#### Previous Line of Lesser Indentation Contains Token

```ruby
class Foo
  def here_is_a_long_method_name(options = {})
    self.bar = 'cool'
    self.bazinga = 'sweet'
  end
end
```

Should be corrected to:

```ruby
class Foo
  def here_is_a_long_method_name(options = {})
    self.bar     = 'cool'
    self.bazinga = 'sweet'
  end
end
```

Instead of:

```ruby
class Foo
  def here_is_a_long_method_name(options = {})
    self.bar                             = 'cool'
    self.bazinga                         = 'sweet'
  end
end
```

#### Previous Line of Greater Indentation Does Not Contain Token

```ruby
class Foo
  def bar(options = {})
    self.bazinga = {
                     one: '1',
                     two: '2',
                   }
    self.bar   = 'cool'
  end
end
```

Should be corrected to:

```ruby
class Foo
  def bar(options = {})
    self.bazinga = {
                     one: '1',
                     two: '2',
                   }
    self.bar     = 'cool'
  end
end
```

Instead of:

```ruby
class Foo
  def bar(options = {})
    self.bazinga = {
                     one: '1',
                     two: '2',
                   }
    self.bar = 'cool'
  end
end
```

**Comment by alexdowad** [comment]
@jfelchner Good points. But I wonder if the suggestion from @Dbz should also be heeded.

```ruby
def method1(arg1, arg2 = :default); end
def method2(arg3 = :default); end
```

It would look pretty strange to align the equals signs in those arglists, don't you think?

**Comment by jfelchner** [comment]
@alexdowad in the rare cases where I have multiple single line methods, I dunno, I might actually like them aligned :grinning: but honestly I could go either way with whatever you decide. :+1:

You've been killing it lately man. Thank you so much for your work.

**Comment by Dbz** [comment]
I agree with Alex that it might seem awkward to align the two methods- especially because you would be aligning the second parameter from `method1` with the first parameter from `method2`.

I also think there might be a bug where calling `rubocop -a` fixes a lot of the equal sign alignment, but still throws the error of not being aligned.

I'd be happy to pair with you @alexdowad on working out a solution to fix these cops if you're up for it. It would give me better insight to the codebase.

**Comment by alexdowad** [comment]
> I'd be happy to pair with you @alexdowad on working out a solution to fix these cops if you're up for it. It would give me better insight to the codebase.

Your suggested fix using `.select` will work fine. (It won't cover all the scenarios presented by @jfelchner, of course.) Would you like help on how to code the body of the `select`?

**Comment by Dbz** [comment]
Oh, I can definitely do the rather simple `select` fix. I was talking about your approach by working upwards and tracking the bug where it says the lines are in violation after fixing them even though they aren't. I'll post a code example this weekend of that.

**Comment by jfelchner** [comment]
@alexdowad would there be any way, while you're in that cop, to make it handle my second use case above?  Where the cop looks at the token placement on the previous line of equal indentation?

**Comment by jfelchner** [comment]
@Dbz I re-read this issue and saw your example for the alignment of those method definitions with fresh eyes.  I didn't even notice that the first one had two arguments hahahaha  Definitely shouldn't align the equal signs on those. :wink:
